reporting/api/src/main/java/org/jboss/windup/reporting/service/ClassificationServiceCache.java

```diff
@@ -0,0 +1,71 @@
+package org.jboss.windup.reporting.service;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.apache.commons.collections.map.LRUMap;
+import org.jboss.windup.config.AbstractRuleLifecycleListener;
+import org.jboss.windup.config.GraphRewrite;
+import org.jboss.windup.config.RuleLifecycleListener;
+import org.jboss.windup.graph.model.resource.FileModel;
+import org.jboss.windup.reporting.model.ClassificationModel;
+
+import com.tinkerpop.blueprints.Vertex;
+import com.tinkerpop.gremlin.java.GremlinPipeline;
+
+/**
+ * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
+ */
+public class ClassificationServiceCache extends AbstractRuleLifecycleListener implements RuleLifecycleListener
```

Comment:
Yes, I think using a Set(fileid) might make the LRU aspect harder. This way, it can make sure that we only cache 15000 or so ids.